aptos-move/e2e-tests/src/executor.rs

```diff
@@ -368,7 +370,19 @@ impl FakeExecutor {
 
     /// Adds an account to this executor's data store.
     pub fn add_account_data(&mut self, account_data: &AccountData) {
-        self.data_store.add_account_data(account_data)
+        self.data_store.add_account_data(account_data);
+        let new_supply = account_data.balance();
+        if new_supply != 0 {
+            let coin_info_resource = self
+                .read_coin_info_resource()
+                .expect("coin info must exist in data store");
+            let old_supply = self.read_coin_supply().unwrap();
+            self.data_store.add_write_set(
+                &coin_info_resource
+                    .to_writeset(old_supply + (new_supply as u128))
+                    .unwrap(),
+            )
+        }
```

Comment:
don't we need an else statement here? feels like a bunch of blind failures otherwsie

Reply:
what's for `else` if newly added supply = 0 we should not have a wrietset for this since no change happens.
But I would add a comment...
(This is an issue found in hacky e2e test setup...